src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -59,6 +59,157 @@ private void InitializeControl()
 
 		internal override bool IsEnabledOverride() => IsEnabled && base.IsEnabledOverride();
 
+		private RoutedEventFlag? _cachedImplementedRoutedEvent;
+
+		internal virtual RoutedEventFlag ImplementedRoutedEvents
+		{
+			get
+			{
+				if (_cachedImplementedRoutedEvent != null)
+				{
+					return _cachedImplementedRoutedEvent.Value;
+				}
+
+				var result = RoutedEventFlag.None;
+				var type = this.GetType();
+
+				var baseClass = type.BaseType;
+				if (baseClass == null || type == typeof(Control))
```

Comment:
⚠️ ⚠️ I removed `typeof(UIElement)` check since it seems unreachable. `Control` inherits from `UIElement`. `GetType` should return the more specific type, so it should never return `UIElement` based on my understanding. Let me know if I'm wrong here.

cc @carldebilly

Reply:
RoutedEvents are not exclusive to `Control`.  Ok some events are only on this type, but not all of them.

Reply:
@carldebilly Can you elaborate more on what you mean here?

Reply:
I mean any FrameworkElement can generate RoutedEvent...  but I just realized only `Control` is providing virtual methods for them.  So you can ignore my comment :-) :-).